### ISSUE TITLE
Web Interface Uses External Libraries - Can't access without internet?

### ISSUE THREAD

**Issue by accessiblepixel** [meshtastic/web#26]
Hi,

I'm a new user of Meshtastic and I've just flashed my first two devices.

To get better access to configuration options I attempted to start the SoftAP on the device and connected my laptop to it. The interface comes up and is just white so I thought I'd check the html source code to see if it was actually getting html.

I've found the reason I'm having issue, obviously connecting directly to a Meshtastic device, you don't have any connection to the internet, and it seems that external requests for JavaScript libraries are required for any functionality on the web front end.

Is there a way to load those libraries onto the device to make it work? I'm unsure how it would otherwise work in a situation that you're offline (one of the primary reasons for the mesh network)?

Kind regards,
Jessica

**Comment by garthvh** [comment]
The hope is to get more of the functionality on the device, currently there is not enough space for some fonts and maps.  Things like maps are probably always going to be too large.

**Comment by accessiblepixel** [comment]
I can completely understand that, but at the minute nothing works if you haven't "preloaded" those files that the interface requires for rendering (a map javascript) and a css theme from a .me address - I'm guessing that at least the stylesheet can be loaded, even if it's just for basic functionality for setting up the device and sending messages?

**Comment by garthvh** [comment]
It does not have offline functionality currently, pull requests for new features are welcome.

**Comment by sachaw** [comment]
I'm aiming to have this fixed with the 1.3 rebuild so wait out for that

**Comment by accessiblepixel** [comment]
@sachaw Awesome! Thank you.

@garthvh If I had any idea how the code was structured and done, I could submit a patch but it's not where my expertise lies and I've only just started with meshtastic and the site said if I found any bugs to report them so I felt that was the thing I could do, at least report what I experienced as a new user.

I do have quite a lot of experience with AVR micro controllers but this is my first interaction with ESP32 + LoRa. The closest I've been is ESP8266 within the Arduino ecosphere, so at the minute documenting what doesn't work is probably the best I can do. 😃
